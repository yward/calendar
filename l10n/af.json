{ "translations": {
    "%s has published the calendar »%s«" : "%s het die kalender »%s« gepubliseer",
    "Hello," : "Dag,",
    "We wanted to inform you that %s has published the calendar »%s«." : "Ons wil u graag inlig dat %s die kalender »%s« gepubliseer het.",
    "Open »%s«" : "Open »%s«",
    "Cheers!" : "Geluk!",
    "Confirm" : "Bevestig",
    "Calendar" : "Kalender",
    "A Calendar app for Nextcloud" : "'n Kalendertoep vir Nextcloud",
    "New event" : "Nuwe geleentheid",
    "Today" : "Vandag",
    "Day" : "Dag",
    "Week" : "Week",
    "Month" : "Maand",
    "List" : "Lys",
    "Preview" : "Voorskou",
    "Copy link" : "Kopieer skakel",
    "Edit" : "Wysig",
    "Delete" : "Skrap",
    "Export" : "Voer uit",
    "Share link" : "Deel skakel",
    "Share with users or groups" : "Deel met gebruikers of groepe",
    "No users or groups" : "Geen gebruikers of groepe",
    "can edit" : "kan wysig",
    "New calendar" : "Nuwe kalender",
    "Name" : "Naam",
    "Deleted" : "Geskrap",
    "Restore" : "Herstel",
    "Delete permanently" : "Skrap permanent",
    "Filename" : "Lêernaam",
    "Cancel" : "Kanselleer",
    "Automatic" : "Outomaties",
    "List view" : "Lysaansig",
    "Actions" : "Aksies",
    "Show week numbers" : "Toon weeknommers",
    "Location" : "Ligging",
    "Description" : "Beskrywing",
    "Duration" : "Duur",
    "to" : "aan",
    "Add" : "Voeg by",
    "Monday" : "Maandag",
    "Tuesday" : "Dinsdag",
    "Wednesday" : "Woensdag",
    "Thursday" : "Donderdag",
    "Friday" : "Vrydag",
    "Saturday" : "Saterdag",
    "Sunday" : "Sondag",
    "Save" : "Stoor",
    "Update" : "Werk by",
    "Your email address" : "U e-posadres",
    "Notification" : "Kennisgewing",
    "Email" : "E-pos",
    "Available" : "Beskikbaar",
    "Not available" : "Onbeskikbaar",
    "Unknown" : "Onbekend",
    "Accept" : "Aanvaar",
    "Tentative" : "Tentatief",
    "All day" : "Heeldag",
    "Repeat" : "Herhaal",
    "never" : "nooit",
    "after" : "na",
    "available" : "beskikbaar",
    "More" : "Nog",
    "Global" : "Globaal",
    "Personal" : "Persoonlik",
    "Details" : "Details",
    "Attendees" : "Bywoners",
    "Resources" : "Hulpbronne",
    "Close" : "Sluit",
    "Anniversary" : "Herdenking",
    "Week {number} of {year}" : "Week {number} van {year}",
    "Daily" : "Daagliks",
    "Weekly" : "Weekliks",
    "Prev" : "Vorige",
    "Next" : "Volgende",
    "Other" : "Ander",
    "Status" : "Status",
    "Confirmed" : "Bevestig",
    "Categories" : "Kategorieë",
    "Commuting" : "In die verkeer",
    "Mail" : "Pos",
    "Birthday" : "Verjaardag"
},"pluralForm" :"nplurals=2; plural=(n != 1);"
}