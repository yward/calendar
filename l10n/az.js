OC.L10N.register(
    "calendar",
    {
    "Cheers!" : "Şərəfə!",
    "Calendar" : "Təqvim",
    "Today" : "Bu gün",
    "Day" : "Gün",
    "Week" : "Həftə",
    "Month" : "Ay",
    "Copy link" : "linki nüsxələ",
    "Edit" : "Dəyişiklik et",
    "Delete" : "Sil",
    "Export" : "Çıxarış",
    "Share link" : "Linki yayımla",
    "can edit" : "dəyişmək olar",
    "New calendar" : "Yeni təqvim",
    "Name" : "Ad",
    "Deleted" : "Silinib",
    "Restore" : "Geri qaytar",
    "Delete permanently" : "Həmişəlik sil",
    "Cancel" : "Dayandır",
    "Automatic" : "Avtomatik",
    "Actions" : "İşlər",
    "Location" : "Yerləşdiyiniz ünvan",
    "Description" : "Açıqlanma",
    "to" : "doğru",
    "Add" : "Əlavə etmək",
    "Monday" : "Bazar ertəsi",
    "Tuesday" : "Çərşənbə axşamı",
    "Wednesday" : "Çərşənbə",
    "Thursday" : "Cümə axşamı",
    "Friday" : "Cümə",
    "Saturday" : "Şənbə",
    "Sunday" : "Bazar",
    "Save" : "Saxla",
    "Update" : "Yenilənmə",
    "Your email address" : "Sizin email ünvanı",
    "Email" : "Email",
    "Send email" : "Email yolla",
    "Repeat" : "Təkrar",
    "never" : "heç vaxt",
    "More" : "Daha da",
    "Personal" : "Şəxsi",
    "Details" : "Detallar",
    "Attendees" : "İştirakçılar",
    "Close" : "Bağla",
    "Daily" : "Günlük",
    "Weekly" : "Həftəlik",
    "Next" : "Növbəti",
    "Other" : "Digər",
    "Mail" : "Məktub",
    "Birthday" : "Ad günü"
},
"nplurals=2; plural=(n != 1);");
