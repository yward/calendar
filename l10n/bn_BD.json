{ "translations": {
    "Cheers!" : "শুভেচ্ছা!",
    "Calendar" : "দিনপঞ্জী",
    "Today" : "আজ",
    "Day" : "দিবস",
    "Week" : "সপ্তাহ",
    "Month" : "মাস",
    "Copy link" : "লিঙ্ক কপি করো",
    "Edit" : "সম্পাদনা",
    "Delete" : "মুছে",
    "Export" : "রপ্তানি",
    "Share link" : "লিংক ভাগাভাগি করেন",
    "can edit" : "সম্পাদনা করতে পারবেন",
    "New calendar" : "নতুন দিনপঞ্জী",
    "Name" : "নাম",
    "Deleted" : "মুছে ফেলা",
    "Restore" : "ফিরিয়ে দাও",
    "Cancel" : "বাতিল",
    "Automatic" : "স্বয়ংক্রিয়",
    "Actions" : "পদক্ষেপসমূহ",
    "Location" : "অবস্থান",
    "Description" : "বিবরণ",
    "to" : "প্রতি",
    "Add" : "যোগ করুন",
    "Monday" : "সোমবার",
    "Tuesday" : "মঙ্গলবার",
    "Wednesday" : "বুধবার",
    "Thursday" : "বৃহস্পতিবার",
    "Friday" : "শুক্রবার",
    "Saturday" : "শনিবার",
    "Sunday" : "রবিবার",
    "Save" : "সংরক্ষণ",
    "Update" : "পরিবর্ধন",
    "Your email address" : "আপনার ই-মেইল ঠিকানা",
    "Email" : "ইমেইল",
    "Unknown" : "অজানা",
    "Send email" : "ইমেইল পাঠান ",
    "Repeat" : "পূনঃসংঘটন",
    "never" : "কখনোই নয়",
    "More" : "বেশী",
    "Personal" : "ব্যক্তিগত",
    "Details" : "বিসতারিত",
    "Attendees" : "অংশগ্রহণকারীবৃন্দ",
    "Close" : "বন্ধ",
    "Daily" : "দৈনিক",
    "Weekly" : "সাপ্তাহিক",
    "Next" : "পরবর্তী",
    "Other" : "অন্যান্য",
    "Mail" : "মেইল",
    "Birthday" : "জন্মদিন"
},"pluralForm" :"nplurals=2; plural=(n != 1);"
}