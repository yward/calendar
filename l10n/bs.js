OC.L10N.register(
    "calendar",
    {
    "Cheers!" : "Cheers!",
    "Calendar" : "Kalendar",
    "Today" : "Danas",
    "Day" : "Dan",
    "Week" : "Sedmica",
    "Month" : "Mjesec",
    "Edit" : "Izmjeni",
    "Delete" : "Obriši",
    "Export" : "Izvezi",
    "Share link" : "Podijelite vezu",
    "can edit" : "mogu mijenjati",
    "New calendar" : "Novi kalendar",
    "Name" : "Ime",
    "Restore" : "Obnovi",
    "Cancel" : "Odustani",
    "Actions" : "Radnje",
    "Location" : "Lokacija",
    "Description" : "Opis",
    "to" : "do",
    "Add" : "Dodaj",
    "Monday" : "Ponedjeljak",
    "Tuesday" : "Utorak",
    "Wednesday" : "Srijeda",
    "Thursday" : "Četvrtak",
    "Friday" : "Petak",
    "Saturday" : "Subota",
    "Sunday" : "Nedjelja",
    "Save" : "Spremi",
    "Update" : "Ažuriraj",
    "Your email address" : "Vaša adresa e-pošte",
    "Email" : "E-pošta",
    "Busy" : "Zauzet",
    "Unknown" : "Nepoznato",
    "Accept" : "Prihvati",
    "Decline" : "Odbij",
    "Send email" : "Pošalji e-poštu",
    "Repeat" : "Ponovi",
    "never" : "nikad",
    "More" : "Više",
    "Personal" : "Osobno",
    "Attendees" : "Sudionici",
    "Close" : "Zatvori",
    "Daily" : "Dnevno",
    "Weekly" : "Sedmično",
    "Next" : "Sljedeće",
    "Other" : "Ostali",
    "Status" : "Status",
    "Mail" : "E-Pošta",
    "Birthday" : "Rođendan"
},
"nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2);");
