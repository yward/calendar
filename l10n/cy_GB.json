{ "translations": {
    "%s has published the calendar »%s«" : "Mae %s wedi cyhoeddi’r calendr » %s«",
    "Hello," : "Helo,",
    "We wanted to inform you that %s has published the calendar »%s«." : "Roeddem am eich hysbysu bod %s wedi cyhoeddi'r calendr » %s«.",
    "Open »%s«" : "Agor »%s«",
    "Cheers!" : "Hwyl!",
    "Calendar" : "Calendr",
    "A Calendar app for Nextcloud" : "Ap Calendr ar gyfer Nextcloud",
    "New event" : "Digwyddiad newydd",
    "Today" : "Heddiw",
    "Day" : "Diwrnod",
    "Week" : "Wythnos",
    "Month" : "Mis",
    "Preview" : "Rhagolwg",
    "Copy link" : "Copïo dolen",
    "Edit" : "Golygu",
    "Delete" : "Dileu",
    "Export" : "Allforio",
    "Share link" : "Rhannu dolen",
    "Share with users or groups" : "Rhannwch gyda defnyddwyr neu grwpiau",
    "No users or groups" : "Dim defnyddwyr na grwpiau",
    "can edit" : "yn gallu golygu",
    "New calendar" : "Calendr newydd",
    "Name" : "Enw",
    "Deleted" : "Wedi dileu",
    "Restore" : "Adfer",
    "Delete permanently" : "Dileu'n barhaol",
    "Cancel" : "Diddymu",
    "List view" : "Golwg rhestr",
    "Actions" : "Gweithredoedd",
    "Show week numbers" : "Dangos rhifau wythnosau",
    "Settings & import" : "Gosodiadau a mewnforio",
    "Location" : "Lleoliad",
    "Description" : "Disgrifiad",
    "to" : "at",
    "Add" : "Ychwanegu",
    "Monday" : "Llun",
    "Tuesday" : "Mawrth",
    "Wednesday" : "Mercher",
    "Thursday" : "Iau",
    "Friday" : "Gwener",
    "Saturday" : "Sadwrn",
    "Sunday" : "Sul",
    "Save" : "Cadw",
    "Update" : "Diweddaru",
    "Email" : "E-bost",
    "Unknown" : "Anhysbys",
    "Tentative" : "I'w gadarnhau",
    "Repeat" : "Ailadrodd",
    "never" : "byth",
    "after" : "wedi",
    "More" : "Rhagor",
    "Global" : "Eang",
    "Personal" : "Personol",
    "Details" : "Manylion",
    "Attendees" : "Mynychwyr",
    "Close" : "Cau",
    "Anniversary" : "Dathliad",
    "Week {number} of {year}" : "Wythnos {number} o {year}",
    "Next" : "Nesaf",
    "Other" : "Arall",
    "When shared show full event" : "Pan wedi ei rannu, yn dangos y digwyddiad llawn",
    "When shared show only busy" : "Pan wedi ei rannu, yn dangos dim ond prysur",
    "When shared hide this event" : "Pan wedi ei rannu, yn cuddio'r digwyddiad hwn",
    "Confirmed" : "Cadarnhawyd",
    "Mail" : "E-bost",
    "Birthday" : "Pen-blwydd"
},"pluralForm" :"nplurals=4; plural=(n==1) ? 0 : (n==2) ? 1 : (n != 8 && n != 11) ? 2 : 3;"
}