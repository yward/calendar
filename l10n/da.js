OC.L10N.register(
    "calendar",
    {
    "User-Session unexpectedly expired" : "Bruger-Sessionen udløb uventet",
    "Provided email-address is not valid" : "Leveret e-mailadresse er ikke gyldig ",
    "%s has published the calendar »%s«" : "%s har oprettet en begivenhed i kalenderen »%s«",
    "Unexpected error sending email. Please contact your administrator." : "Uventet fejl ved afsendelse, venligst kontakt din administrator",
    "Hello," : "Hej,",
    "We wanted to inform you that %s has published the calendar »%s«." : "Vi ønkser at informerer dig om at %s har oprettet en begivenhed i kalenderen »%s«.",
    "Open »%s«" : "Åbn »%s«",
    "Cheers!" : "Hav en fortsat god dag.",
    "Appointments" : "Aftaler",
    "Confirm" : "Bekræft",
    "Date:" : "Dato:",
    "Where:" : "Hvor:",
    "Description:" : "Beskrivelse:",
    "Calendar" : "Kalender",
    "A Calendar app for Nextcloud" : "En kalender-app til Nextcloud",
    "Previous day" : "Forrige dag",
    "Previous week" : "Forrige uge",
    "Previous month" : "Forrige måned",
    "Next day" : "Næste dag",
    "Next week" : "Næste uge",
    "Next month" : "Næste måned",
    "New event" : "Opret begivenhed",
    "Today" : "I dag",
    "Day" : "Dag",
    "Week" : "Uge",
    "Month" : "Måned",
    "List" : "Liste",
    "Preview" : "Forhåndsvisning",
    "Copy link" : "Kopiér link",
    "Edit" : "Rediger",
    "Delete" : "Slet",
    "Untitled calendar" : "Unanvngiven kalender",
    "Edit name" : "Rediger navn",
    "Saving name …" : "Gemmer navn…",
    "Edit color" : "Rediger farve",
    "Saving color …" : "Gemmer farve…",
    "Copy private link" : "Kopier privat link",
    "Export" : "Eksportér",
    "Unshare from me" : "Fjern deling fra mig",
    "An error occurred, unable to change visibility of the calendar." : "Kalenderens synlighed kunne ikke ændres.",
    "An error occurred, unable to delete the calendar." : "Kalenderen kunne ikke slettes.",
    "Calendar link copied to clipboard." : "Kalender link kopieret.",
    "Calendar link could not be copied to clipboard." : "Kalender link kunne ikke kopieres.",
    "An error occurred, unable to rename the calendar." : "Kalenderen kunne ikke omdøbes.",
    "An error occurred, unable to change the calendar's color." : "Kalenderens farve kunne ikke ændres.",
    "_Deleting the calendar in {countdown} second_::_Deleting the calendar in {countdown} seconds_" : ["Kalenderen bliver slettet om {countdown} sekunder","Kalenderen bliver slettet om {countdown} sekunder"],
    "Share link" : "Del link",
    "Publish calendar" : "Udgiv kalender",
    "Publishing calendar" : "Udgiver kalender",
    "Copy public link" : "Kopier offentligt link",
    "Send link to calendar via email" : "Send link til kalender via e-mail",
    "Sending email …" : "Sender email…",
    "Copy subscription link" : "Kopier abonnementslink",
    "Copying link …" : "Kopierer link…",
    "Copied link" : "Link kopieret",
    "Could not copy link" : "Link kunne ikke kopieres",
    "Copy embedding code" : "Kopier indlejringskode",
    "Copying code …" : "Kopierer kode…",
    "Copied code" : "Kode kopieret",
    "Could not copy code" : "Kode kunne ikke kopieres",
    "Delete share link" : "Slet delingslink",
    "Deleting share link …" : "Sletter delingslink…",
    "An error occurred, unable to publish calendar." : "Kalenderen kunne ikke udgives",
    "Embed code copied to clipboard." : "Indlejringskode kopieret til udklipsholder.",
    "Share with users or groups" : "Del med brugere eller grupper",
    "No users or groups" : "Ingen brugere eller grupper",
    "can edit" : "kan redigere",
    "+ New calendar" : "+ Ny kalender",
    "New calendar" : "Ny kalender",
    "Creating calendar …" : "Opretter kalender…",
    "New calendar with task list" : "Ny kalender med opgaveliste",
    "Creating subscription …" : "Opretter abonnement…",
    "Name" : "Navn",
    "Deleted" : "Slettet",
    "Restore" : "Gendan",
    "Delete permanently" : "Slet permanent",
    "Empty trash bin" : "Tom papirkurv",
    "Import calendars" : "Importer kalendere",
    "Filename" : "Filnavn",
    "Cancel" : "Annullér",
    "_Import calendar_::_Import calendars_" : ["Importer kalender","Importer kalendere"],
    "Automatic" : "Automatisk",
    "or" : "eller",
    "List view" : "Vis som liste",
    "Actions" : "Handlinger",
    "Enable birthday calendar" : "Slå fødselsdagskalender til",
    "Enable simplified editor" : "Slå simpel editor til",
    "Show weekends" : "Vis weekender",
    "Show week numbers" : "Vis ugenummer ",
    "Copy primary CalDAV address" : "Kopier primær CalDAV-adresse",
    "Show keyboard shortcuts" : "Vis tastaturgenveje",
    "Settings & import" : "Indstillinger & importer",
    "No reminder" : "Ingen påmindelse",
    "Location" : "Sted",
    "Description" : "Beskrivelse",
    "Visibility" : "Synlighed",
    "Duration" : "Varighed",
    "to" : "til",
    "Add" : "Tilføj",
    "Monday" : "Mandag",
    "Tuesday" : "Tirsdag",
    "Wednesday" : "Onsdag",
    "Thursday" : "Torsdag",
    "Friday" : "Fredag",
    "Saturday" : "Lørdag",
    "Sunday" : "Søndag",
    "Save" : "Gem",
    "Update" : "Opdatér",
    "Your email address" : "Din email adresse",
    "Notification" : "Notifikation",
    "Email" : "E-mail",
    "_second_::_seconds_" : ["sekund","sekunder"],
    "_minute_::_minutes_" : ["minut","minutter"],
    "_hour_::_hours_" : ["time","timer"],
    "_day_::_days_" : ["dag","dage"],
    "_week_::_weeks_" : ["uge","uger"],
    "Available" : "Tilgængelig",
    "Busy" : "Optaget",
    "Unknown" : "Ukendt",
    "Accept" : "Accepter",
    "Decline" : "Afvis",
    "Tentative" : "Foreløbig",
    "Create Talk room for this event" : "Opret Snak rum for denne begivenhed",
    "Send email" : "Send e-mail",
    "Event title" : "Titel",
    "All day" : "Hele dagen",
    "from {startDate}" : "fra {startDate}",
    "from {startDate} at {startTime}" : "fra {startDate} kl {startTime}",
    "to {endDate}" : "til {endDate}",
    "to {endDate} at {endTime}" : "til {endDate} kl {endTime}",
    "Repeat" : "Gentag",
    "never" : "aldrig",
    "after" : "efter",
    "first" : "første",
    "third" : "tredje",
    "fourth" : "fjerde",
    "fifth" : "femte",
    "second to last" : "næstsidste",
    "last" : "sidste",
    "_month_::_months_" : ["måned","måneder"],
    "_year_::_years_" : ["år","år"],
    "weekday" : "hverdag",
    "weekend day" : "weekenddag",
    "available" : "tilgængelig",
    "More" : "Mere",
    "Pick a date" : "Vælg en dato",
    "Global" : "Global",
    "Time:" : "Tid:",
    "Personal" : "Personlig",
    "Details" : "Detaljer",
    "Attendees" : "Deltagere",
    "Resources" : "Resourcer",
    "Close" : "Luk",
    "Anniversary" : "Årsdag",
    "Week {number} of {year}" : "Uge {number} i {year}",
    "Daily" : "Dagligt",
    "Weekly" : "Ugentligt",
    "Untitled event" : "Unavngiven begivenhed",
    "Next" : "Næste",
    "Year" : "År",
    "Other" : "Andet",
    "When shared show full event" : "Når delt, vis den fulde begivenhed",
    "When shared show only busy" : "Når delt, vis kun optaget",
    "When shared hide this event" : "Når delt, skjul denne begivenhed",
    "Status" : "Status",
    "Confirmed" : "Bekræftet",
    "Canceled" : "Annulleret",
    "Categories" : "Kategorier",
    "Presentation" : "Præsentation",
    "Talk" : "Snak",
    "Movie" : "Film",
    "Review" : "Gennemse",
    "Office" : "Kontor",
    "Mail" : "Mail",
    "Birthday" : "Fødselsdag",
    "User not found" : "Bruger ikke fundet"
},
"nplurals=2; plural=(n != 1);");
