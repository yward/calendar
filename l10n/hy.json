{ "translations": {
    "Cheers!" : "Չը՛խկ",
    "Calendar" : "Օրացույց",
    "New event" : "Նոր իրադարձություն",
    "Today" : "Այսօր",
    "Day" : "Օր",
    "Week" : "Շաբաթ",
    "Month" : "Ամիս",
    "Copy link" : "Պատճենել հղումը",
    "Edit" : "մշակել",
    "Delete" : "Ջնջել",
    "Export" : "Արտահանում",
    "Share link" : "Կիսվել հղմամբ",
    "Share with users or groups" : "Կիսվել օգտատերերի կամ խմբերի հետ",
    "can edit" : "կարող է խմբագրել",
    "Name" : "Անուն",
    "Deleted" : "Ջնջված",
    "Restore" : "Վերականգնել",
    "Delete permanently" : "Ջնջել ընդմիշտ",
    "Cancel" : "Չեղարկել",
    "Location" : "Տեղակայություն",
    "Description" : "Նկարագրություն",
    "Add" : "Ավելացնել",
    "Monday" : "Երկուշաբթի",
    "Tuesday" : "Երեքշաբթի",
    "Wednesday" : "Չորեքշաբթի",
    "Thursday" : "Հինգշաբթի",
    "Friday" : "Ուրբաթ",
    "Saturday" : "Շաբաթ",
    "Sunday" : "Կիրակի",
    "Save" : "Պահպանել",
    "Update" : "Թարմացնել",
    "Email" : "Էլ․փոստ",
    "Unknown" : "Անհայտ",
    "Repeat" : "Կրկնել",
    "never" : "երբեք",
    "after" : "հետո",
    "More" : "Ավելի",
    "Global" : "Ընդհանուր",
    "Personal" : "Անձնական",
    "Details" : "Մանրամասներ",
    "Attendees" : "Մասնակիցներ",
    "Close" : "Փակել",
    "Next" : "Հաջորդ",
    "Other" : "Այլ",
    "Confirmed" : "Հաստատված",
    "Mail" : "Փոստ",
    "Birthday" : "Ծննդյան օր"
},"pluralForm" :"nplurals=2; plural=(n != 1);"
}