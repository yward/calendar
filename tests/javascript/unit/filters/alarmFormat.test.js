/**
 * @copyright Copyright (c) 2019 Georg Ehrke
 *
 * @author Georg Ehrke <oc.list@georgehrke.com>
 *
 * @license GNU AGPL version 3 or any later version
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 *
 */
import alarmFormat from "../../../../src/filters/alarmFormat.js";
import { translate, translatePlural } from '@nextcloud/l10n'

jest.mock('@nextcloud/l10n')

describe('format/alarmFormat test suite', () => {

	beforeEach(() => {
		translate.mockClear()
		translatePlural.mockClear()

		translate
			.mockImplementation((app, str) => str)

		translatePlural
			.mockImplementation((app, singStr, pluStr) => pluStr)
	})

	it('should format an alarm for an all-day event at midnight', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: true,
			relativeIsRelatedToStart: true,
			relativeAmountTimed: 0,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 0,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 0,
			relativeMinutesAllDay: 0,
			relativeTrigger: 0,
		}

		expect(alarmFormat(alarm, true, 'Europe/Berlin', 'de')).toEqual('Midnight on the day the event starts')
	})

	it('should format an alarm for an all-day event days before', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: true,
			relativeIsRelatedToStart: true,
			relativeAmountTimed: 0,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 1,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 9,
			relativeMinutesAllDay: 0,
			relativeTrigger: -15 * 60 * 60,
		}

		expect(alarmFormat(alarm, true, 'Europe/Berlin', 'de')).toMatchSnapshot()
	})

	it('should format an alarm for an all-day event weeks weeks before', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: true,
			relativeIsRelatedToStart: true,
			relativeAmountTimed: 0,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 1,
			relativeUnitAllDay: 'weeks',
			relativeHoursAllDay: 9,
			relativeMinutesAllDay: 0,
			relativeTrigger: -159 * 60 * 60 - 30 * 60,
		}

		expect(alarmFormat(alarm, true, 'Europe/Berlin', 'de')).toMatchSnapshot()
	})

	it('should format an alarm for an all-day event on the same day at a certain time', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: false,
			relativeIsRelatedToStart: true,
			relativeAmountTimed: 0,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 0,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 9,
			relativeMinutesAllDay: 0,
			relativeTrigger: 32400,
		}

		expect(alarmFormat(alarm, true, 'Europe/Berlin', 'de')).toMatchSnapshot()
	})

	it('should format an alarm for an all-day event not supported in the default range', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: false,
			relativeIsRelatedToStart: true,
			relativeAmountTimed: 33,
			relativeUnitTimed: 'hours',
			relativeAmountAllDay: 1,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 9,
			relativeMinutesAllDay: 0,
			relativeTrigger: 118800,
		}

		expect(alarmFormat(alarm, true, 'Europe/Berlin', 'de')).toMatchSnapshot()
	})

	it('should format a relative trigger at the events start', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: true,
			relativeIsRelatedToStart: true,
			relativeAmountTimed: 0,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 0,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 0,
			relativeMinutesAllDay: 0,
			relativeTrigger: 0,
		}

		expect(alarmFormat(alarm, false, 'Europe/Berlin', 'de')).toEqual('at the event\'s start')
	})

	it('should format a relative trigger at the events end', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: true,
			relativeIsRelatedToStart: false,
			relativeAmountTimed: 0,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 0,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 0,
			relativeMinutesAllDay: 0,
			relativeTrigger: 0,
		}

		expect(alarmFormat(alarm, false, 'Europe/Berlin', 'de')).toEqual('at the event\'s end')
	})

	it('should format a relative trigger before the event starts', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: true,
			relativeIsRelatedToStart: true,
			relativeAmountTimed: 15,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 0,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 0,
			relativeMinutesAllDay: 0,
			relativeTrigger: -900,
		}

		expect(alarmFormat(alarm, false, 'Europe/Berlin', 'de')).toMatchSnapshot()
	})

	it('should format a relative trigger before the event ends', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: true,
			relativeIsRelatedToStart: false,
			relativeAmountTimed: 15,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 0,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 0,
			relativeMinutesAllDay: 0,
			relativeTrigger: -900,
		}

		expect(alarmFormat(alarm, false, 'Europe/Berlin', 'de')).toMatchSnapshot()
	})

	it('should format a relative trigger after the event starts', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: false,
			relativeIsRelatedToStart: true,
			relativeAmountTimed: 15,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 0,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 0,
			relativeMinutesAllDay: 0,
			relativeTrigger: 900,
		}

		expect(alarmFormat(alarm, false, 'Europe/Berlin', 'de')).toEqual('{time} after the event starts')
	})

	it('should format a relative trigger after the event ends', () => {
		const alarm = {
			type: 'EMAIL',
			isRelative: true,
			absoluteDate: null,
			absoluteTimezoneId: null,
			relativeIsBefore: false,
			relativeIsRelatedToStart: false,
			relativeAmountTimed: 15,
			relativeUnitTimed: 'minutes',
			relativeAmountAllDay: 0,
			relativeUnitAllDay: 'days',
			relativeHoursAllDay: 0,
			relativeMinutesAllDay: 0,
			relativeTrigger: 900,
		}

		expect(alarmFormat(alarm, false, 'Europe/Berlin', 'de')).toEqual('{time} after the event ends')
	})

	it('should format an absolute alarm in the user\'s timezone', () => {
		const date = new Date(2019, 0, 1, 0, 0, 0, 0)
		const alarm = {
			type: 'EMAIL',
			isRelative: false,
			absoluteDate: date,
			absoluteTimezoneId: 'Europe/Berlin',
			relativeIsBefore: null,
			relativeIsRelatedToStart: null,
			relativeAmountTimed: null,
			relativeUnitTimed: null,
			relativeAmountAllDay: null,
			relativeUnitAllDay: null,
			relativeHoursAllDay: null,
			relativeMinutesAllDay: null,
			relativeTrigger: null,
		}

		expect(alarmFormat(alarm, false, 'Europe/Berlin', 'de')).toEqual('on {time}')
	})

	it('should format an absolute alarm in a different timezone', () => {
		const date = new Date(2019, 0, 1, 0, 0, 0, 0)
		const alarm = {
			type: 'EMAIL',
			isRelative: false,
			absoluteDate: date,
			absoluteTimezoneId: 'America/New_York',
			relativeIsBefore: null,
			relativeIsRelatedToStart: null,
			relativeAmountTimed: null,
			relativeUnitTimed: null,
			relativeAmountAllDay: null,
			relativeUnitAllDay: null,
			relativeHoursAllDay: null,
			relativeMinutesAllDay: null,
			relativeTrigger: null,
		}

		expect(alarmFormat(alarm, true, 'Europe/Berlin', 'de')).toEqual('on {time} ({timezoneId})')
	})

})
