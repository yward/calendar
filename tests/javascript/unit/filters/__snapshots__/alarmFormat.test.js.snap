// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format/alarmFormat test suite should format a relative trigger before the event ends 1`] = `"{time} before the event ends"`;

exports[`format/alarmFormat test suite should format a relative trigger before the event starts 1`] = `"{time} before the event starts"`;

exports[`format/alarmFormat test suite should format an alarm for an all-day event days before 1`] = `"%n days before the event at {formattedHourMinute}"`;

exports[`format/alarmFormat test suite should format an alarm for an all-day event not supported in the default range 1`] = `"{time} after the event starts"`;

exports[`format/alarmFormat test suite should format an alarm for an all-day event on the same day at a certain time 1`] = `"on the day of the event at {formattedHourMinute}"`;

exports[`format/alarmFormat test suite should format an alarm for an all-day event weeks weeks before 1`] = `"%n weeks before the event at {formattedHourMinute}"`;
